Cesium.CodeGen/Contexts/TranslationUnitContext.cs

```diff
@@ -75,30 +99,84 @@ internal IType ResolveType(IType type)
             {
                 if (_tags.TryGetValue(structType.Identifier, out var existingType))
                 {
-                    return existingType;
+                    resolvedType = existingType;
+                    unresolvedType = null;
+                    return true;
                 }
             }
 
-            var members = structType.Members
-                .Select(structMember => structMember with { Type = ResolveType(structMember.Type) })
-                .ToList();
-            return new StructType(members, structType.Identifier);
+            bool isStructResolved = true;
+            var members = new List<LocalDeclarationInfo>();
+            unresolvedType = null;
+            foreach (var structMember in structType.Members)
+            {
+                if (TryResolveType(structMember.Type, out var memberResolvedType, out var memberUnresolvedType))
+                {
+                    members.Add(structMember with { Type = memberResolvedType });
+                }
+                else
+                {
+                    if (unresolvedType is null)
+                    {
+                        unresolvedType = memberResolvedType;
+                    }
+
+                    members.Add(structMember);
+                }
+            }
+
+            resolvedType = new StructType(members, structType.Identifier);
+            return isStructResolved;
         }
 
         if (type is FunctionType functionType)
         {
+            bool isFunctionResolved = true;
```

Comment:
Ok, so there are a lot of warnings on that the variable is unused. Could you remove it then?